src/main/java/io/nats/client/impl/OrderedMessageManager.java

```diff
@@ -67,13 +68,25 @@ protected ManageResult manage(Message msg) {
         return manageStatus(msg);
     }
 
+    @Override
+    protected void handleHeartbeatError() {
+        super.handleHeartbeatError();
+        handleErrorCondition();
+    }
+
     private void handleErrorCondition() {
         try {
             targetSid.set(null);
             expectedExternalConsumerSeq = 1; // consumer always starts with consumer sequence 1
 
-            // 1. shutdown the manager, for instance stops heartbeat timers
-            shutdown();
+            // 1. delete the consumer by name so we can recreate it with a different delivery policy
+            //    b/c we cannot edit a push consumer's delivery policy
+            JetStreamManagement jsm = conn.jetStreamManagement(js.jso);
+            String actualConsumerName = sub.getConsumerName();
+            try {
+                jsm.deleteConsumer(stream, actualConsumerName);
+            }
+            catch (Exception ignore) {}
```

Comment:
So on failure of (push) ordered, I want to re-use the same consumer name, but it's likely that the original consumer I was using is still around, I don't know why the ordered error was triggered. But because you cannot just update an existing consumer to change deliver policy/start sequence, I have to make an entire new consumer. So I just make sure I delete the old consumer. The times its does exception will most likely be that it actually was deleted and deleteConsumer threw a JSApi exception